xrpl/models/amounts/issued_currency_amount.py

```diff
@@ -25,3 +28,34 @@ class IssuedCurrencyAmount(IssuedCurrency):
 
     :meta hide-value:
     """
+
+    @classmethod
+    def from_issued_currency(
+        cls: Type[IssuedCurrencyAmount], issued_currency: Currency, value: str
+    ) -> Union[IssuedCurrencyAmount, str]:
+        """
+        Build an IssuedCurrencyAmount from an IssuedCurrency.
+
+        Args:
+            issued_currency: The issued currency to use.
+            value: The amount of issued currency.
+
+        Returns:
+            An Amount with the provided value in the provided issued
+            currency.
+        """
+        if isinstance(issued_currency, XRP):
+            return value
+        return cast(
+            IssuedCurrencyAmount,
+            cls.from_dict({**issued_currency.to_dict(), "value": value}),
+        )
+
+    def get_issued_currency(self: IssuedCurrencyAmount) -> IssuedCurrency:
```

Comment:
what about `to_issued_currency` instead?

Reply:
actually, `to_currency`. then your instance method on `Currency` can be `to_amount`